### PR TITLE
Dynamic names support

### PR DESCRIPTION
**Pull request by adam-fowler**

### CODE REVIEW

File: Sources/Mustache/Template+Parser.swift [hunk 1]
```diff
@@ -478,7 +508,7 @@ extension MustacheTemplate {
         return state.newLine && self.hasLineFinished(&parser)
     }
 
-    private static let sectionNameCharsWithoutBrackets = Set<Character>("0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ.-_?")
-    private static let sectionNameChars = Set<Character>("0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ.-_?()")
+    private static let sectionNameCharsWithoutBrackets = Set<Character>("0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ.-_?*")
```
Comment (Joannis) [inline Sources/Mustache/Template+Parser.swift:511]: What's this change for?

Comment (Joannis) [reply Sources/Mustache/Template+Parser.swift:511]: I'd assume it's for a `*` prefixed section name. Is that the only allowed use case of `*`?

**Review by Joannis — APPROVED** [review]

**Comment by Joannis** [comment]
Looks good, I think the chars allowed makes sense but I'll let you double-check that

File: Sources/Mustache/Template+Parser.swift (hunk 1 above)
Comment (adam-fowler) [reply Sources/Mustache/Template+Parser.swift:511]: Yes it is

**Pull request merged**